tools/storage-advisor/src/s3Issues.go

```diff
@@ -373,68 +531,181 @@ func runS3Checks(cfg aws.Config, err error) {
 
-	/*
+	checkEncryption(buckets, client, &issues)
 
-		checkInventory(buckets, client)
+	checkLogging(buckets, client, &issues)
 
-		for i := range buckets {
-			// Get value of bucket's server access logging
-			r, err := client.GetBucketLogging(context.TODO(), &s3.GetBucketLoggingInput{
-				Bucket: &buckets[i].Name,
-			})
-			if err != nil {
-				buckets[i].Logging = err.Error()
-			} else {
-				if r.LoggingEnabled == nil {
-					buckets[i].Logging = "disabled"
-				} else {
-					buckets[i].Logging = fmt.Sprintf("%s/%s", *r.LoggingEnabled.TargetBucket, *r.LoggingEnabled.TargetPrefix)
-				}
-			}
+	checkInventory(buckets, client, &issues)
+
+	writeBasicIssues(issues, identity)
+
+	if api.IsConfigured() {
+		uploadBasicIssues(api, buckets)
+	}
+
+	return buckets
+}
+
+func writeBasicIssues(issues []FindingWithBucket, identity string) {
+	// Group issues by buckets
+	issuesByBucket := make(map[string][]FindingWithBucket)
+	for _, issue := range issues {
+		issuesByBucket[issue.Bucket] = append(issuesByBucket[issue.Bucket], issue)
+	}
+
+	f, err := os.Create("s3-recommendations.txt")
+	if err != nil {
+		fmt.Printf("Cannot write issues list: cannot open 's3-recommendations.txt': %s\n", err.Error())
+		return
+	}
+	defer f.Close()
+
+	fmt.Fprintf(f, "\nS3 issue check was run as %s\n", identity)
+
+	if len(issues) == 0 {
+		fmt.Fprint(f, "\nThere are not issues. You rock!\n")
+		return
+	}
+
+	fmt.Fprint(f, "\nPlease consider making the following changes:\n\n")
+
+	for bucket, issues := range issuesByBucket {
+		fmt.Fprintf(f, "%s:\n", bucket)
+		for _, issue := range issues {
+			fmt.Fprintf(f, "  - %s\n", issue.Recommentation)
+		}
+		fmt.Fprintf(f, "\n")
+	}
+	fmt.Println(green("Wrote recommentations to 's3-recommentations.txt'\n"))
+
+	fmt.Println("We can write recommendations in Terraform format, which might be easier to apply.")
+	fmt.Println("Note that we just write Terraform code, we'll never try to change anything in your AWS account.")
+
+	prompt := promptui.Select{
+		Label: "Proceed>",
+		Items: []string{"Yes, write Terraform recommendations", "No, skip Terraform recommendations"},
+	}
+
+	index, result, err := prompt.Run()
+	if err != nil {
+		fmt.Printf("Could not get the prompt: %s\n", err.Error())
+		return
+	}
+	if index == 1 {
+		return
+	}
+
+	fmt.Println("We'll need to know which bucket is good for S3 logs and inventory.")
+
+	validate := func(input string) error {
+		if input == "" {
+			return errors.New("Bucket name cannot be empty")
+		}
+		si := strings.Index(input, "/")
+		if si == -1 {
+			return errors.New("Please specify both bucket and prefix, separated by '/'")
 		}
+		if si == len(input)-1 {
+			return errors.New("Prefix is empty")
+		}
+		if si < 3 {
+			return errors.New("Bucket name is too short")
+		}
+		return nil
+	}
+	prompt2 := promptui.Prompt{
+		Label:    "Specify the bucket/prefix for logging and inventory",
+		Validate: validate,
+	}
+	result, _ = prompt2.Run()
+	si := strings.Index(result, "/")
```

Comment:
There is `strings.Cut` for exactly this reason.

Reply:
Thanks!